modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -64,48 +70,109 @@ const Diagram = ({
 		radius: newPinSettings.defaultRadius,
 	});
 
-	useEffect(() => {
-		fetch(`${pinsEndpoint}${productId}/pins`)
+	const loadPins = useCallback(
+		() =>
+			fetch(`${pinsEndpoint}${PRODUCTS}/${productId}/${PINS}`, {
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+			})
+				.then((response) => response.json())
+				.then((jsonResponse) => {
+					const loadedPins = jsonResponse.items.map((item) => ({
+						cx: item.positionX,
+						cy: item.positionY,
+						id: item.id,
+						label: item.number,
+					}));
+
+					setCpins(loadedPins);
+				}),
+		[pinsEndpoint, productId]
+	);
+
+	const deletePin = (node) => {
+		fetch(`${pinsEndpoint}${PINS}/${node.id}`, {
+			headers: new Headers({
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			}),
+			method: 'DELETE',
+		});
+
+		// toast "pins deleted"
+
+	};
+
+	const updatePin = (node) => {
+		if (node.id !== undefined) {
+			fetch(`${pinsEndpoint}${PINS}/${node.id}`, {
+				body: JSON.stringify(node),
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+				method: 'PATCH',
+			}).then((response) => {
+				response.json();
+			});
+		}
+		else {
+			fetch(`${pinsEndpoint}${PRODUCTS}/${productId}/${PINS}`, {
+				body: JSON.stringify(node),
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+				method: 'POST',
+			}).then((response) => {
+				response.json();
```

Comment:
What is the purpose of these json responses if you aren't doing anything with it or returning it? I could be missing something here.